### PR TITLE
Silence warnings without DEBUG flag

### PR DESCRIPTION
**Pull request by vadim-gleyf**
## what
* Add a feature flag to make it possible to disable warnings

## why
* Warnings are great for debugging, but scary for users

## references
- https://github.com/cloudposse/build-harness/issues/89

### CODE REVIEW

**Comment by osterman** [comment]
LGTM! Please output with debug enabled and debug disabled.

**Review by osterman — CHANGES_REQUESTED** [review]
See comments. Post output in comments from test runs.

File: modules/git/bootstrap.Makefile [hunk 1]
```diff
@@ -27,3 +28,4 @@ else
 endif
 
 endif
+endif
```
Comment (aknysh) [inline modules/git/bootstrap.Makefile]: is it correct that if `d` flag is not set, we skip the entire file?
do we want to disable just the warning?

```
ifeq ($(wildcard .git),)
  ifneq (,$(findstring d,$(MAKEFLAGS)))
      $(warning disabling git bootstrapping)
  endif
else
.....
endif
```

**Review by aknysh — COMMENTED** [review]
see comments

File: modules/git/bootstrap.Makefile (hunk 1 above)
Comment (osterman) [reply modules/git/bootstrap.Makefile]: @aknysh good catch! That does look like the case. @vadim-hleif We only want to disable the warnings, not the logic =)

**Review by aknysh — APPROVED** [review]

**Comment by vadim-gleyf** [comment]
capture

- Just copied part of code for demonstrating

**Review by osterman — APPROVED** [review]

**Pull request merged**